system/Debug/Toolbar/Views/toolbarloader.js.php

```diff
@@ -9,45 +12,43 @@ function loadDoc(time) {
+	let url = "<?= rtrim(site_url(), '/') ?>";
+	let xhttp = new XMLHttpRequest();
 
-	var xhttp = new XMLHttpRequest();
 	xhttp.onreadystatechange = function() {
 		if (this.readyState === 4 && this.status === 200) {
-			var toolbar = document.getElementById("toolbarContainer");
-			if (!toolbar) {
+			let toolbar = document.getElementById("toolbarContainer");
+
+			if (! toolbar) {
 				toolbar = document.createElement('div');
 				toolbar.setAttribute('id', 'toolbarContainer');
 				document.body.appendChild(toolbar);
 			}
 
-			// copy for easier manipulation
 			let responseText = this.responseText;
+			let dynamicStyle = document.getElementById('debugbar_dynamic_style');
+			let dynamicScript = document.getElementById('debugbar_dynamic_script');
 
-			// get csp blocked parts
-			// the style block is the first and starts at 0
-			{
-				let PosBeg = responseText.indexOf( '>', responseText.indexOf( '<style' ) ) + 1;
-				let PosEnd = responseText.indexOf( '</style>', PosBeg );
-				document.getElementById( 'debugbar_dynamic_style' ).innerHTML = responseText.substr( PosBeg, PosEnd - PosBeg );
-				responseText = responseText.substr( PosEnd + 8 );
-			}
-			// the script block starts right after style blocks ended
-			{
-				let PosBeg = responseText.indexOf( '>', responseText.indexOf( '<script' ) ) + 1;
-				let PosEnd = responseText.indexOf( '</script>' );
-				document.getElementById( 'debugbar_dynamic_script' ).innerHTML = responseText.substr( PosBeg, PosEnd - PosBeg );
-				responseText = responseText.substr( PosEnd + 9 );
-			}
-			// check for last style block
-			{
-				let PosBeg = responseText.indexOf( '>', responseText.lastIndexOf( '<style' ) ) + 1;
-				let PosEnd = responseText.indexOf( '</style>', PosBeg );
-				document.getElementById( 'debugbar_dynamic_style' ).innerHTML += responseText.substr( PosBeg, PosEnd - PosBeg );
-				responseText = responseText.substr( 0, PosBeg + 8 );
-			}
+			// get the first style block, copy contents to dynamic_style, then remove here
+			let start = responseText.indexOf('>', responseText.indexOf('<style')) + 1;
+			let end = responseText.indexOf('</style>', start);
+			dynamicStyle.innerHTML = responseText.substr(start, end - start);
+			responseText = responseText.substr(end + 8);
+
+			// get the first script after the first style, copy contents to dynamic_script, then remove here
+			start = responseText.indexOf('>', responseText.indexOf('<script')) + 1;
+			end = responseText.indexOf('\<\/script>', start);
+			dynamicScript.innerHTML = responseText.substr(start, end - start);
+			responseText = responseText.substr(end + 9);
+
+			// check for last style block, append contents to dynamic_style, then remove here
+			start = responseText.indexOf('>', responseText.indexOf('<style')) + 1;
+			end = responseText.indexOf('</style>', start);
+			dynamicStyle.innerHTML += responseText.substr(start, end - start);
+			responseText = responseText.substr(0, start - 8);
```

Comment:
before: `responseText = responseText.substr( 0, PosBeg + 8 );`
after: `responseText = responseText.substr(0, start - 8);`

Why did the `+` change to `-`?

Reply:
In the `before` the substr will include the `<style>` tag. This change ensures it doesn't

Reply:
I got it. It is where the bug existed.